### ISSUE TITLE
Blank mp4

### ISSUE THREAD

**Issue by braidp1** [skiwithpete/timelapsepi#2]
The script works great, but it creates a blank mp4.  I modified the script to make everything else was working and can create an animated gif before the pics are removed so I have located the issue with the mp4 conversion line I believe.  Maybe a am missing additional installs?  Thanks

**Comment by skiwithpete** [comment]
Are avconv and libx264 installed and working?

**Comment by braidp1** [comment]
How do I check that?  I have installed all sorts of packages I have read about in the tutorial so I thought they were in there.  
When I write in sudo apt-get install libx264 it says it is unable to locate package.
Thanks

**Comment by braidp1** [comment]
For example, when I write the same for libav-tools it says it is already the newest version

**Comment by skiwithpete** [comment]
What happens when you run this line? 

avconv -r 24 -i image%04d.jpg -vcodec libx264 -crf 20 -g 15 `date +%Y%m%d%H%M`timelapse.mp4

do you get error messages?

(note you will need a couple of files named image0000.jpg, image0001.jpg etc

**Comment by braidp1** [comment]
I am not sure how to run that seperately, but I do not get any errors from the python script when I run it.  If I comment out the picture removal I see the the camera takes and puts the pics in the right folder.  The animated gif script creates the correct one too.  The mp4 creates in correct folder, there just is not anything in it...no errors.

If I can test this seperately I will need to know how to do that.  Thanks !

**Comment by braidp1** [comment]
I do not have ffmpeg if that makes a difference...says it is from another source/no installation candidate.  I did install ...tools-extra with no change.  Unless I need to reboot or something

**Comment by skiwithpete** [comment]
run it, simply by pasting it into terminal

**Comment by braidp1** [comment]
It is telling me there is such image in directory

**Comment by skiwithpete** [comment]
?

Can you paste the results?

Again you will need to create some jpgs with the right name to put in the
directory you want to run this from.

**Comment by braidp1** [comment]
I have files image0000.jpg, image0001.jpg, image0002.jpg, and image0003.jpg in same directory as everything else, /home/pi/timelapse .

This is what says in the terminal
pi@raspberrypi:~ $ avconv -r 24 -i image%04d.jpg -vcodec libx264 -crf 20 -g 15 date +%Y%m%d%H%Mtimelapse.mp4
avconv version 11.11-6:11.11-1~deb8u1+rpi1, Copyright (c) 2000-2017 the Libav developers
  built on Nov  2 2017 13:07:45 with gcc 4.9.2 (Raspbian 4.9.2-10)
image%04d.jpg: No such file or directory

And I tried it with a name of a specific image with same results
pi@raspberrypi:~ $ avconv -r 24 -i image0000.jpg -vcodec libx264 -crf 20 -g 15 date +%Y%m%d%H%Mtimelapse.mp4
avconv version 11.11-6:11.11-1~deb8u1+rpi1, Copyright (c) 2000-2017 the Libav developers
  built on Nov  2 2017 13:07:45 with gcc 4.9.2 (Raspbian 4.9.2-10)
image0000.jpg: No such file or directory

**Comment by skiwithpete** [comment]
Ok, you need to put image0000.jpg in your home directory.  As in: /home/pi

Right now, the only error is that it can't find the file.

**Comment by braidp1** [comment]
Put them in the correct directory and got

pi@raspberrypi:~ $ avconv -r 24 -i image0000.jpg -vcodec libx264 -crf 20 -g 15 date +%Y%m%d%H%Mtimelapse.mp4
avconv version 11.11-6:11.11-1~deb8u1+rpi1, Copyright (c) 2000-2017 the Libav developers
  built on Nov  2 2017 13:07:45 with gcc 4.9.2 (Raspbian 4.9.2-10)
Input #0, image2, from 'image0000.jpg':
  Duration: 00:00:00.04, start: 0.000000, bitrate: N/A
    Stream #0.0: Video: mjpeg, yuvj420p, 1027x768, 25 tbn
Unable to find a suitable output format for 'date'

**Comment by skiwithpete** [comment]
Sorry, that's python code... paste this revised instead

avconv -r 24 -i image0000.jpg -vcodec libx264 -crf 20 -g 15 test.mp4

**Comment by braidp1** [comment]
It left a test mp4 in the directory with nothing in the video

pi@raspberrypi:~ $ avconv -r 24 -i image0000.jpg -vcodec libx264 -crf 20 -g 15 test.mp4
avconv version 11.11-6:11.11-1~deb8u1+rpi1, Copyright (c) 2000-2017 the Libav developers
  built on Nov  2 2017 13:07:45 with gcc 4.9.2 (Raspbian 4.9.2-10)
Input #0, image2, from 'image0000.jpg':
  Duration: 00:00:00.04, start: 0.000000, bitrate: N/A
    Stream #0.0: Video: mjpeg, yuvj420p, 1027x768, 25 tbn
File 'test.mp4' already exists. Overwrite ? [y/N]

**Comment by skiwithpete** [comment]
ok, so no errors with the script.  Just that the MP4 seems blank.

Can you put the MP4 on another system?  I mean, it should have one frame
which would take 0.5seconds to play... But it shouldn't produce an error.

**Comment by braidp1** [comment]
I emailed my animated gif to myself and that played on my phone.

When I tried to email this file gmail said it would not attached it because it was 0kb.  When I tried to play it with VLC on pi it does not show anything, even for sec.

**Comment by skiwithpete** [comment]
But VLC doesn't show an error.

I think the problem is that the sample is too small.  You should create an
MP4 with more pics.  Hundreds, if not thousands, and try again.

**Comment by braidp1** [comment]
Although commented out the shutdown, when I tried 300 images and came back the screen was black and I could not wake up the pi.  I unplugged it and rebooted.  The mp4 was blank but maybe it was computing and I unplugged it too soon.

So I ran it again and kept the pi awake during the entire capture period and it locked up while I was on it. I am going to leave it for a while in case it is computing, then I will check on mp4.  Has been locked about 10min and I can see usage was at 90 percent when froze.

Pi has one red and one yellow light on if that means anything.

**Comment by braidp1** [comment]
After 2hours the screen did not go black, but the screen was locked up so I I had to unplug the pi and reboot.  The folder had all 300 pics and the mp4 with nothing in it.  I thought it might have got caught on the animated gif so I commented that out and ran it with 100.  It ran through until the end and removed the pics but still had the blank mp4.

I ran again with the gif at 200.  Everything ran thru and removed the pics with 0kb mp4.

Put animated gif back in and ran it, but seemed like that was bogging system down.

I noticed I did not have your ratio in there for resolution or the one suggested for gif.  I think it was getting caught up on that.  I put ratio of 1024 x 768 at 50 frames and both worked!  I see this was only 2sec so I tried it 100, which also worked with an mp4 at 4 sec.  I can see that the sampling and videos for mp4 and gif are different both worked without locking up at lower and proper ratio!

Thanks for all of your help, way more than expected.
